server/utils.go

```diff
@@ -56,35 +56,20 @@ func mergeEnvs(imageConfig *v1.Image, kubeEnvs []*types.KeyValue) []string {
 		}
 
 		if imageConfig != nil {
-			for _, imageEnv := range imageConfig.Config.Env {
-				var found bool
-
-				parts := strings.SplitN(imageEnv, "=", 2)
-				if len(parts) != 2 {
-					continue
-				}
-
-				imageEnvKey := parts[0]
-				if imageEnvKey == "" {
-					continue
+			// Pre-compute env keys map for O(1) lookup instead of O(M) repeated splits
+			envKeys := make(map[string]bool, len(envs))
```

Comment:
nit: we can use set though the difference in memory consumption is almost negligible.

```suggestion
			envKeys := make(map[string]struct{}, len(envs))
```